builtin/providers/azurerm/resource_arm_resource_group.go

```diff
@@ -0,0 +1,162 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/azure-sdk-for-go/arm/resources"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmResourceGroup returns the *schema.Resource
+// associated to resource group resources on ARM.
+func resourceArmResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmResourceGroupCreate,
+		Read:   resourceArmResourceGroupRead,
+		Exists: resourceArmResourceGroupExists,
+		Delete: resourceArmResourceGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmResourceGroupName,
```

Comment:
Hm I don't remember that particular exchange, but we're definitely pushing as much validation as we can up to the  `terraform plan` stage so users receive feedback there.

Perhaps that old conversation was about maintaining a list of valid values from within Terraform I was worried might change?

Anyways, the overarching direction today is most definitely "as much validation as we can reasonably do up front" :+1: